### PR TITLE
[merp] Print missing status marker file for stage 1 (setup)

### PR DESCRIPTION
**Pull request by alexischr**
Simplify by moving all marker file creation to `mono_summarize_timeline_phase_log` 
/cc @lambdageek @kdubau

### CODE REVIEW

File: mono/utils/mono-state.c [hunk 1]
```diff
@@ -170,6 +170,10 @@ mono_summarize_timeline_start (void)
 
 	log.level = MonoSummarySetup;
 	log.directory = configured_timeline_dir;
+	char out_file [200];
+	file_for_summary_stage (log.directory, log.level, out_file, sizeof(out_file));
+	int handle = g_open (out_file, O_WRONLY | O_CREAT, S_IWUSR | S_IRUSR | S_IRGRP | S_IROTH);
+	close(handle);
```
Comment (lambdageek) [inline mono/utils/mono-state.c]: Make a helper for this function and for `mono_summarize_timeline_phase_log` that just does this bit.

**Comment by alexischr** [comment]
@monojenkins backport to 2019-10
@monojenkins backport to 2019-08

File: mono/utils/mono-state.c [hunk 2]
```diff
@@ -148,6 +148,15 @@ file_for_summary_stage (const char *directory, MonoSummaryStage stage, gchar *bu
 	g_snprintf (buff, sizeof_buff, "%s%scrash_stage_%d", directory, G_DIR_SEPARATOR_S, stage);
 }
 
+static void
+create_stage_mark_file (void)
+{
+	char out_file [200];
+	file_for_summary_stage (log.directory, log.level, out_file, sizeof(out_file));
+	int handle = g_open (out_file, O_WRONLY | O_CREAT, S_IWUSR | S_IRUSR | S_IRGRP | S_IROTH);
+	close(handle);
```
Comment (lambdageek) [inline mono/utils/mono-state.c]: Coverity didn't like that we call `close (handle)` without checking that `handle > 0`.

```
>>>     CID 1454754:  Error handling issues  (NEGATIVE_RETURNS)
>>>     "handle" is passed to a parameter that cannot be negative.
```

Comment (alexischr) [reply mono/utils/mono-state.c]: I'll push a fix. That pattern was already there though, is Coverity only looking at diffs?

Comment (lambdageek) [reply mono/utils/mono-state.c]: The website has a report for the whole codebase,  but it only sends me emails about changed code.

**Pull request merged**